src/tasks/MonoTargetsTasks/ILStrip/ILStrip.cs

```diff
@@ -342,27 +339,29 @@ private static int ComputeMethodHash(PEReader peReader, int rva)
         return (peReader.PEHeaders.SectionHeaders[sectionIndex].PointerToRawData + relativeOffset);
     }
 
-    private static int ComputeMethodHeaderSize(MemoryStream memStream, int actualLoc)
+    private static int ComputeMethodHeaderSize(Stream stream, int actualLoc)
     {
-        memStream.Position = actualLoc;
-        int firstbyte = memStream.ReadByte();
+        stream.Position = actualLoc;
+        int firstbyte = stream.ReadByte();
         int headerFlag = firstbyte & 0b11;
         return (headerFlag == 2 ? 1 : 4);
     }
 
-    private static void SetCodeSizeToZeroForTiny(MemoryStream memStream, int actualLoc)
+    private static void SetCodeSizeToZeroForTiny(Stream stream, int actualLoc)
     {
-        memStream.Position = actualLoc;
-        memStream.WriteByte(0b10);
+        stream.Position = actualLoc;
+        stream.WriteByte(0b10);
+        stream.Flush();
     }
 
-    private static void ZeroOutMethodBody(MemoryStream memStream, int methodSize, int actualLoc, int headerSize)
+    private static void ZeroOutMethodBody(Stream stream, int methodSize, int actualLoc, int headerSize)
     {
-        memStream.Position = actualLoc + headerSize;
+        stream.Position = actualLoc + headerSize;
         byte[] zeroBuffer;
-        zeroBuffer = ArrayPool<byte>.Shared.Rent(methodSize);
+        zeroBuffer = ArrayPool<byte>.Shared.Rent(methodSize - headerSize);
         Array.Clear(zeroBuffer, 0, zeroBuffer.Length);
-        memStream.Write(zeroBuffer, 0, methodSize - headerSize);
+        stream.Write(zeroBuffer, 0, zeroBuffer.Length);
+        stream.Flush();
         ArrayPool<byte>.Shared.Return(zeroBuffer);
```

Comment:
I haven't checked but would this be faster? it probably takes advantage of caches

```
for (int i = 0; i < methodSize - headerSize; i++)
  stream.WriteByte(0);
```

also don't need to flush here